cordova-lib/src/cordova/platform.js

```diff
@@ -174,6 +181,24 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
     });
 }
 
+function save(hooksRunner, projectRoot, opts) {
+    var xml = cordova_util.projectConfig(projectRoot);
+    var cfg = new ConfigParser(xml);
+
+    // First, remove all platforms that are already in config.xml
+    cfg.getEngines().map(function(engine){
+        cfg.removeEngine();
+    });
+
+    // Save installed platforms into config.xml
+    return platformMetadata.getPlatformVersions(projectRoot).then(function(platformVersions){
+        platformVersions.map(function(platVer){
```

Comment:
Changed all the map into forEach. it's definitely better to use forEach with side-effecting operations.